### PR TITLE
Add missing material theme string widgets

### PR DESCRIPTION
**Pull request by stanlemon**
### Reasons for making this change

Several string widgets are missing from the material theme, specifically color, date, date time, email and URI. Without these widgets using the various formats for fields with type 'string' will fall back to the bootstrap theme. This PR adds support for all of those widgets in material UI with the same reuse pattern as in the bootstrap theme (pass back the `TextWidget` with a more specific type, see https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/src/components/widgets/ColorWidget.js for an example).

_Note that I needed to add 'type' to the `WidgetProps` to support this. The bootstrap widgets do not have their props typed so I believe this is why that wasn't an issue before._

Here is an example schema that you can use in the playground to see what happens when you use the material theme.

```json
{
  "title": "String Widgets",
  "description": "String Widgets",
  "type": "object",
  "properties": {
    "date": {
      "type": "string",
      "title": "Date Widget",
      "format": "date"
    },
    "dateTime": {
      "type": "string",
      "title": "Date Time Widget",
      "format": "date-time"
    },
    "email": {
      "type": "string",
      "title": "Email Widget",
      "format": "email"
    },
    "uri": {
      "type": "string",
      "title": "URI Widget",
      "format": "uri"
    },
    "color": {
      "type": "string",
      "title": "Color Widget",
      "format": "color"
    }
  }
}

```

There is a similar PR just for the date widget in #1765 

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by stanlemon** [comment]
I don't think this requires any documentation changes since this is just filling in gaps on the material theme, but I am happy to add anything if desired. As far as tests, I wasn't entirely sure what was best to add here. There are existing tests in `StringField_test.js` but they are not material-specific. Any guidance there would be appreciated.

**Comment by stanlemon** [comment]
Snapshots were updated for email and url, I will add some schema for the other types.

I just noticed that my prettier extension in vscode reformatted some of the stuff I touched. I didn't find a prettier config, I did notice this comment https://react-jsonschema-form.readthedocs.io/en/latest/contributing/#coding-style and just wanted to make sure the defaults are being used and some of the formatting stuff included here is OK or if I should back it out.

**Comment by stanlemon** [comment]
Tests added and snapshots have been updated @epicfaace.

**Comment by stanlemon** [comment]
Interesting, looks like that happens in Chrome but not Safari. It appears to be a bug with material-ui. From what I gather I need to set this to make the empty render not look ugly:

```
InputLabelProps={{
  shrink: true
}}
```

On the `TextWidget` for that one. That will mean that label will not bounce up above the input upon focus, which is the current default behavior on rjsf. If you're OK with that I'll work that in.

Thinking out loud here... We could use these pickers https://material-ui.com/components/pickers/ for the date and time widgets, or we could use them in the alt-date alt-datetime widgets too. Any interest in either of those options?

**Comment by stanlemon** [comment]
@epicfaace here's a set of changes that would allow for passing the shrink settings. https://github.com/stanlemon/react-jsonschema-form/pull/1/files

It also allows for further properties to be passed down to the underlying `TextWidget` which could come in handy. Let me know if this seems agreeable and I'll merge it into this branch.

**Comment by epicfaace** [comment]
> Thinking out loud here... We could use these pickers https://material-ui.com/components/pickers/ for the date and time widgets, or we could use them in the alt-date alt-datetime widgets too. Any interest in either of those options?

Yeah, those would probably be ideal. We would need to add a peer dependency on `@material-ui/pickers`, though, so let's not do that in this PR. Let's save that for the next major version.

**Comment by stanlemon** [comment]
> Yeah, those would probably be ideal. We would need to add a peer dependency on `@material-ui/pickers`, though, so let's not do that in this PR. Let's save that for the next major version.

Just to clarify: They would be ideal for the main widgets or the alt widgets?

**Comment by epicfaace** [comment]
> > Yeah, those would probably be ideal. We would need to add a peer dependency on `@material-ui/pickers`, though, so let's not do that in this PR. Let's save that for the next major version.
> 
> Just to clarify: They would be ideal for the main widgets or the alt widgets?

Probably for the main widgets.

**Comment by epicfaace** [comment]
> This will mean that label will not bounce up above the input upon focus, which is the current default behavior on rjsf. If you're OK with that I'll work that in.

Sounds good to me!

**Comment by stanlemon** [comment]
Good deal. I merged in the change to support the shrink param, so that issue should be resolved.

**Comment by stanlemon** [comment]
Clean build, this one might be ready. 🤞 If you have any guidance on that last commit with regards to the typing let me know. I think it is OK, but my typescript-foo is limited so I'm not entirely confident.

**Comment by epicfaace** [comment]
@stanlemon can you also change the format in which the date widget gives its value?

Right now, the format doesn't seem to be valid for a "date-time" format in JSON Schema:

image


Compare this with the default date widget value:

image

**Comment by stanlemon** [comment]
@epicfaace good catch, 81b20350329a2197a3ae1c607a5026a3cea88d25 should resolve that. Thanks.

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -1048,6 +1048,36 @@ export function toDateString(
   return time ? datetime : datetime.slice(0, 10);
 }
 
+export function utcToLocal(jsonDate) {
```
Comment (epicfaace) [inline packages/core/src/utils.js:1051]: What exactly is the point of this function?

Comment (stanlemon) [reply packages/core/src/utils.js:1051]: This is what ensures the format is correct in the Bootstrap theme. I didn’t write this, just simply relocated it to utils.

**Pull request merged**